lib/i18n_getstrings.dart

```diff
@@ -55,17 +93,26 @@ class StringExtractor extends UnifyingAstVisitor<void> {
 
   @override
   void visitSimpleStringLiteral(SimpleStringLiteral node) {
-    if (_hasI18nSyntax(node, node.parent!)) {
-      strings.add(node.value);
+    final DecodedSintax sintax = _hasI18nSyntax(node, node.parent!);
+    if (sintax.valid) {
+      final ExtractedString s = ExtractedString(node.value,
+          pluralRequired:
+              sintax.modifiers.contains(I18nRequiredModifiers.plural));
+      strings.add(s);
     }
 
     return super.visitSimpleStringLiteral(node);
   }
 
   @override
   void visitAdjacentStrings(AdjacentStrings node) {
-    if (_hasI18nSyntax(node.strings.last, node.parent!)) {
-      strings.add(node.stringValue!);
+    final DecodedSintax sintax =
```

Comment:
There is too much duplicated code now. Please factor this and the same code above out into a new method, e.g. `_handleI18nSyntax`.